Shared/sdk/CVector2D.h

```diff
@@ -20,128 +20,123 @@
 class CVector2D
 {
 public:
-    CVector2D()
-    {
-        fX = 0;
-        fY = 0;
-    }
+    float fX = 0.0f;
+    float fY = 0.0f;
+
+    constexpr CVector2D() noexcept = default;
 
-    CVector2D(float _fX, float _fY)
+    constexpr CVector2D(float x, float y) noexcept :
+        fX(x),
+        fY(y)
     {
-        fX = _fX;
-        fY = _fY;
     }
 
-    CVector2D(const CVector& vec)
+    constexpr CVector2D(const CVector& vec) noexcept :
+        fX(vec.fX),
+        fY(vec.fY)
     {
-        fX = vec.fX;
-        fY = vec.fY;
     }
 
-    CVector2D(const CVector4D& vec)
+    constexpr CVector2D(const CVector4D& vec) noexcept :
+        fX(vec.fX),
+        fY(vec.fY)
     {
-        fX = vec.fX;
-        fY = vec.fY;
     }
 
-    CVector2D& operator=(const CVector& vec)
+
+    constexpr CVector2D& operator=(const CVector& vec) noexcept
     {
         fX = vec.fX;
         fY = vec.fY;
         return *this;
     }
 
-    CVector2D& operator=(const CVector4D& vec)
+    constexpr CVector2D& operator=(const CVector4D& vec) noexcept
     {
         fX = vec.fX;
         fY = vec.fY;
         return *this;
     }
 
-    float DotProduct(CVector2D& other) const { return fX * other.fX + fY * other.fY; }
+    constexpr float DotProduct(CVector2D& other) const { return fX * other.fX + fY * other.fY; }
 
-    float Length() const { return sqrt(fX * fX + fY * fY); }
+    float Length() const { return std::hypotf(fX, fY); }
 
-    float LengthSquared() const { return (fX * fX) + (fY * fY); }
+    // LengthSquared returns Length() without sqrt applied (i.e. returns x*x* + y*y)
+    // For more info see CMaterialLine3DBatcher::Flush()
```

Comment:
```suggestion
    // LengthSquared returns Length() without sqrt applied (i.e. returns x*x* + y*y).
    // This can be useful if you only want to compare lengths.
```